src/main/java/com/minecolonies/coremod/colony/workorders/AbstractWorkOrder.java

```diff
@@ -1,536 +1,766 @@
-                 .getBuildings()
-                 .values()
-                 .stream()
-                 .anyMatch(building -> building instanceof BuildingBuilder && !building.getAllAssignedCitizen().isEmpty() && building.getBuildingLevel() >= level);
-    }
-
-    /**
-     * Check if this workOrder can be resolved by an existing builder by distance.
-     *
-     * @param colony the colony to check in.
-     * @param level  the new level of the building.
-     * @return true if so.
-     */
-    @Override
-    public boolean tooFarFromAnyBuilder(final IColony colony, final int level)
-    {
-        return false;
-    }
-}
+package com.minecolonies.coremod.colony.workorders;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.buildings.IBuilding;
+import com.minecolonies.api.colony.workorders.IWorkManager;
+import com.minecolonies.api.colony.workorders.IWorkOrder;
+import com.minecolonies.api.colony.workorders.IWorkOrderView;
+import com.minecolonies.api.colony.workorders.WorkOrderType;
+import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.api.util.Log;
+import com.minecolonies.api.util.Tuple;
+import com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingBuilder;
+import com.minecolonies.coremod.colony.workorders.view.AbstractWorkOrderView;
+import com.minecolonies.coremod.colony.workorders.view.WorkOrderBuildingView;
+import com.minecolonies.coremod.colony.workorders.view.WorkOrderDecorationView;
+import com.minecolonies.coremod.colony.workorders.view.WorkOrderMinerView;
+import com.minecolonies.coremod.tileentities.TileEntityDecorationController;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TranslationTextComponent;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.minecolonies.api.util.constant.Suppression.UNUSED_METHOD_PARAMETERS_SHOULD_BE_REMOVED;
+
+/**
+ * General information between WorkOrders.
+ */
+public abstract class AbstractWorkOrder implements IWorkOrder
+{
+    /**
+     * NBT for storage.
+     */
+    private static final String TAG_TYPE                = "type";
+    private static final String TAG_ID                  = "id";
+    private static final String TAG_TH_PRIORITY         = "priority";
+    private static final String TAG_CLAIMED_BY          = "claimedBy";
+    private static final String TAG_CLAIMED_BY_BUILDING = "claimedByBuilding";
+    private static final String TAG_STRUCTURE_NAME      = "structureName";

```

Comment:
Is this a good idea? Shouldn't the constants only contain globally identical tag names, almost all of these tags aren't shared (with the exception of something like type/ID).
There's a potential problem also where if someone accidently edits the wrong constant it could mess up the serialization.

Reply:
Could also be a WorkOrderConstants class.

Reply:
> Could also be a WorkOrderConstants class.

Is this a new preferred way vs just defining the constants in the class that needs them? I personally do not see the point of exposing them to other classes if only ever 1 class needs them.